src/library/autodj/autodjprocessor.cpp

```diff
@@ -297,14 +291,18 @@ void AutoDJProcessor::fadeNow() {
         fadeTime = spinboxTime;
     }
 
-    double timeUntilEndOfFromTrack = pFromDeck->trackTime() - fromDeckCurrentPosition;
+    double timeUntilEndOfFromTrack = fromDeckEndPosition - fromDeckCurrentPosition;
     fadeTime = math_min(fadeTime, timeUntilEndOfFromTrack);
     pFromDeck->fadeEndPos = fromDeckCurrentPosition + fadeTime;
 
     // These are expected to be a fraction of the track length.
-    pFromDeck->fadeBeginPos /= pFromDeck->trackTime();
-    pFromDeck->fadeEndPos /= pFromDeck->trackTime();
-    pToDeck->startPos /= pToDeck->trackTime();
+    pFromDeck->fadeBeginPos /= fromDeckEndPosition;
```

Comment:
Dividing by an absolute position is conceptually wrong. Please rethink the naming.

Reply:
I have changed the name from trackTime() because this was ambiguous: current vs total. 
Now it is clear that it IS the end position, with a matching name to the other positions used in this function all measured in time from 0:00 taking the current rate into account. 

This line transforms fadeBeginPos form a temporary seconds resolution to the ratio  0...1 we need to compare time independent later. 

It actually a bad concept to store both in the same variable. 
Unfortunately changing this is quite error prone and IMO not worth the risk. 

I am happy to change the naming or add comments if you have some ideas though.

Reply:
As someone who as not familiar with the actual implementation I tried to follow the code. These calculations don't make sense to me with the current naming. The plain numbers may be correct, but their representation is confusing and unsound.

Reply:
Store a single value for consistency internally and provide functions for the transformation to different domains and dimensions.